src/components/cards/ConstructBuilding.tsx

```diff
@@ -1,123 +0,0 @@
-import React, { Dispatch, SetStateAction } from "react";
-import {
-  Buildings,
-  BuildingType,
-  ResourcePool,
-  Resources,
-  ResourceType,
-} from "../../types/";
-import {
-  cloneBasicObjectWithJSON,
-  updateResources as updateResourcePool,
-} from "../../utils";
-import { resourceChecker } from "../../utils/resourceChecker";
-import { AddRemoveButton } from "../buttons";
-import {
-  CardCostsInfo,
-  CardHeader,
-  CardShowCount,
-  CardSymbol,
-  CardTemplate,
-} from "../cards";
-import CardImageHeaderDescriptionContainer from "./CardImageHeaderDescriptionContainer";
-
-interface ConstructBuildingProps {
-  buildings: Buildings;
-  setBuildings: Dispatch<SetStateAction<Buildings>>;
-  /* setBuildings: (buildings: Buildings) => void; */
-  /* FIXME: Why is this being inferred as a number? */
-  buildingType: BuildingType;
-  resources: Resources;
-  setResources: Dispatch<SetStateAction<Resources>>;
-  resourcePool: ResourcePool;
-  setResourcePool: Dispatch<SetStateAction<ResourcePool>>;
-  /* setResources: (resources: Resources) => void; */
-}
-
-export default function ConstructBuilding({
-  buildings,
-  setBuildings,
-  buildingType,
```

Comment:
So one thing I noticed down below. You are always doing `buildings[buildingType]`. You can do one of two things here

1. `const building = buildings[buildingType]`. Do it once at the top and then just use it.
2. Don't pass in both `buildings` and `buildingType`. Just take a `building`. Let the parent component do `<ConstructBuilding building={buildings[buildingType]}` instead. Prefer this because theres no need for this component to know about all the rest of the buildings.